src/new-client/src/App/LiveRates/Tiles/ExecutionStatusAlertSwitch/Done.tsx

```diff
@@ -0,0 +1,76 @@
+import { format } from "date-fns"
+import { useCurrencyPair } from "services/currencyPairs"
+import { useExecution, onExecutionDismiss, ExecutionStatus } from "services/executions"
+import { formatNumber } from "utils"
+import { Button, CurrencyPairDiv, ExecutionStatusAlertContainer, TradeIdDiv, TradeMessageDiv } from "./styled"
+import styled from "styled-components/macro"
+import { Direction } from "services/trades"
+import { useBaseTerm } from "../TileHeader"
+import { useContext } from "react"
+import { SymbolContext } from "../Tile"
+
+export const BackgroundColored = styled.span`
+  background-color: ${({ theme }) => theme.textColor};
+  color: ${({ theme }) => theme.colors.accents.positive.base};
+  padding: 0.25rem;
+`
+
+const pastTenseDirection = (d: Direction): string => {
+    if (d === Direction.Buy) {
+        return 'bought'
+    }
+
+    if (d === Direction.Sell) {
+        return 'sold'
+    }
+
+    return "Never" as never
+}
+
+const buildTradeMessage = (
+    base: string,
+    direction: Direction,
+    notional: number,
+    spotRate: number,
+    terms: string,
+    valueDate: string
+) =>
+    <TradeMessageDiv>
+        {`You ${pastTenseDirection(direction)} `}
+        <BackgroundColored>{`${base} ${formatNumber(notional)}`}</BackgroundColored>
+        {` at a rate of `}
+        <BackgroundColored>{`${spotRate}`}</BackgroundColored>
+        {` for ${terms} ${formatNumber(notional * spotRate)}`}
+        {` settling (SPT) ${format(new Date(valueDate), "dd-MMM")}`}
+    </TradeMessageDiv>
+
+const Done = () => {
+    const symbol = useContext(SymbolContext)
+    const {
+        currencyPair,
+        direction,
+        id,
+        notional,
+        spotRate,
+        tradeId,
+        valueDate,
+    } = useExecution(symbol)
+    const baseTerm = useBaseTerm(symbol)
+
+    const closeAlert = () => {
+        onExecutionDismiss({ currencyPair, id })
+    }
+
+    const { base, terms } = useCurrencyPair(symbol)
+
+    return (
+        <ExecutionStatusAlertContainer status={ExecutionStatus.Done}>
```

Comment:
Feel like you could easily abstract the shell for Done/Rejected into a common component.  Then Done/Rejected render that component and supply the dynamic info

```
        <ExecutionStatusAlertContainer status={FROM Done/Rejected}>
            <CurrencyPairDiv>{baseTerm}</CurrencyPairDiv>
            <TradeIdDiv>{`Trade Id: ${tradeId}`}</TradeIdDiv>
            <TradeMessageDiv>{ children (from Done/Rejected }</TradeMessageDiv>
            <Button onClick={closeAlert} success={FROM Done/Rejected}>Dismiss</Button>
        </ExecutionStatusAlertContainer>
```